doc/motion_planning_pipeline/src/motion_planning_pipeline_tutorial.cpp

```diff
@@ -64,14 +65,27 @@ int main(int argc, char** argv)
   //
   // .. _RobotModelLoader:
   //     http://docs.ros.org/indigo/api/moveit_ros_planning/html/classrobot__model__loader_1_1RobotModelLoader.html
-  robot_model_loader::RobotModelLoader robot_model_loader("robot_description");
-  robot_model::RobotModelPtr robot_model = robot_model_loader.getModel();
+  robot_model_loader::RobotModelLoaderPtr robot_model_loader(
+      new robot_model_loader::RobotModelLoader("robot_description"));
+  robot_model::RobotModelPtr robot_model = robot_model_loader->getModel();
 
   // Using the :moveit_core:`RobotModel`, we can construct a
   // :planning_scene:`PlanningScene` that maintains the state of
   // the world (including the robot).
   planning_scene::PlanningScenePtr planning_scene(new planning_scene::PlanningScene(robot_model));
 
+  /* Create a RobotState and JointModelGroup to keep track of the current robot pose and planning group*/
+  robot_state::RobotState& robot_state = planning_scene->getCurrentStateNonConst();
+  const robot_model::JointModelGroup* joint_model_group = robot_state.getJointModelGroup("panda_arm");
+
+  // With the planning scene we create a planing scene monitor that
+  // monitors planning scene diffs and applys them to the planning scene
+  planning_scene_monitor::PlanningSceneMonitorPtr psm(
+      new planning_scene_monitor::PlanningSceneMonitor(planning_scene, robot_model_loader));
+  psm->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE);
+  psm->startStateMonitor();
```

Comment:
add comments documenting what each of these ``start()`` functions do. I always find they are named ambiguously and are confusing.